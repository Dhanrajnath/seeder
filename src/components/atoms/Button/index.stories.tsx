import { Meta, StoryObj } from '@storybook/react';
import DeleteIcon from '@mui/icons-material/Delete';
import { ButtonProps, PaletteColor } from '@mui/material';
import Button from '.';
import theme from '../../../utils/theme';

export default {
    title: 'Atoms/Button',
    component: Button,
    argTypes: {
        onClick: { action: 'clicked' },
    },
} as Meta;

type ButtonType = StoryObj<ButtonProps>;


export const Primary: ButtonType = {
    args: {
        size: 'large',
        variant: 'contained',
        color: 'primary',
        children: 'Primary Button',
        onClick: () => { console.log('Clicked!') }
    }
};

export const DeleteButton: ButtonType = {
    args: {
        size: 'large',
        variant: 'outlined',
        color: 'error',
        children: 'DELETE',
        startIcon: <DeleteIcon />
    }
}

export const SubmitButton: ButtonType = {
    args: {
        size: 'large',
        variant: 'text',
        color: 'success',
        children: 'SUBMIT'
    }
}

export const NewCashKickButton: ButtonType = {
    args: {
        children: 'New Cash Kick',
        sx: {
            ...theme.typography.button,
            borderRadius: theme.spacing(3),
            background: theme.palette.primary["500" as keyof PaletteColor],
            gap: theme.spacing(2),
            padding: '20px 40px 20px 40px',
            width: theme.spacing(69),
            height: theme.spacing(15)
        }
    }
}

export const MyContractsButton: ButtonType = {
    args: {
        variant: 'contained',
        children: 'My Contracts',
        sx: {
            ...theme.typography.button,
            border: '1px',
            borderRadius: theme.spacing(3),
            borderColor: theme.palette.primary["500" as keyof PaletteColor],
            background: theme.palette.primary["600" as keyof PaletteColor],
            gap: theme.spacing(2),
            padding: '12px 24px 12px 24px',
            width: theme.spacing(38),
            height: theme.spacing(11)
        }
    }
}

export const MyCachKicksButton: ButtonType = {
    args: {
        variant: 'contained',
        children: 'My Cash Kicks',
        sx: {
            ...theme.typography.button,
            border: '1px',
            borderRadius: theme.spacing(3),
            borderColor: theme.palette.primary["500" as keyof PaletteColor],
            background: theme.palette.primary["600" as keyof PaletteColor],
            gap: theme.spacing(2),
            padding: '12px 24px 12px 24px',
            width: theme.spacing(39),
            height: theme.spacing(11)
        }
    }
}
