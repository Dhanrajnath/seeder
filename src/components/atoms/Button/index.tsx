import { Button as MUIButton, ButtonProps, styled } from "@mui/material";
import theme from "../../../utils/theme";

const StyledButton = styled(MUIButton)(theme.typography.button);

const Button = ({
    variant = 'contained',
    children,
    ...rest
}: ButtonProps) => {
    return (
        <StyledButton variant={variant}  {...rest}>
            {children}
        </StyledButton>
    );
};

export default Button;