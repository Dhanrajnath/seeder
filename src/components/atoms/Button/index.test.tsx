import { fireEvent, render } from "@testing-library/react";
import Button from ".";

describe('button tests', () => {
    const mockOnClick = jest.fn();

    it('renders a contained button', () => {
        const { container, getByRole, getByText } = render(<Button onClick={mockOnClick}>Submit</Button>);
        const button = getByRole('button');

        expect(button).toHaveClass('MuiButton-contained');

        fireEvent.click(button);
        expect(mockOnClick).toHaveBeenCalled();

        const buttonText = getByText('Submit');
        expect(buttonText).toBeInTheDocument();

        expect(container.firstChild).toMatchSnapshot();
    });

    it('renders a outlined button', () => {
        const { container } = render(<Button variant="outlined" onClick={mockOnClick}>Submit</Button>);

        expect(container.firstChild).toMatchSnapshot();
    });

});