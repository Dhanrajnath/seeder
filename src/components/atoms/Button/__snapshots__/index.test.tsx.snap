// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button tests renders a contained button 1`] = `
<button
  class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-1twgh2p-MuiButtonBase-root-MuiButton-root"
  tabindex="0"
  type="button"
>
  Submit
  <span
    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
  />
</button>
`;

exports[`button tests renders a outlined button 1`] = `
<button
  class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-h4y63c-MuiButtonBase-root-MuiButton-root"
  tabindex="0"
  type="button"
>
  Submit
  <span
    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
  />
</button>
`;
